### DISCUSSION TITLE
Looking for a way to swap out the config for the current window

### DISCUSSION THREAD

**Discussion by notbenh** [wezterm/wezterm#6360] [General]
I'm using the wez-tmux plugin to replace kitty + tmux, it works great for local but I have a remote development server that I use mosh to keep a long running tmux session. I am looking for a way to have a single `dev-server` script that will connect with an alternate leader set.

So far I've not found a way to issue config changes from the CLI. I found the toggle leader stuff but that's assigning a key to trigger the event rather than have an event triggered from a script and all of `wezterm cli` is about pane and window management rather than issuing actions.

So far the best I have pulled together is to make a copy of my config for the development server and can connect via `wezterm --config-file ~/.config/wezterm/devserver.lua start -- mosh dev-server tmux attach` but that starts a new window and leaves me with an unusable pane.

Any ways to approach this better or have a config swap in place?

**Comment by bew** [comment]
Have you tried `window:set_config_overrides` method?
https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html 

To trigger the action, checkout this guide to pass data from the pane to your config:
https://wezfurlong.org/wezterm/recipes/passing-data.html

**Comment by notbenh** [comment]
It's possible I'm not understating how that would relate but neither of those seem to address my issue. Ultimately I want to have a shell script that in effect looks something like:

```
#!/bin/bash
wezterm change config.leader to { key = 'b', mods = 'CTRL' }
mosh dev-server tmux attach
wezterm change config.leader to { key = 'a', mods = 'CTRL' }
```

What I don't want, as stated in my post, is to have to remember a rarely uses hot key combination just to change the leader so that I can then just call mosh and then have to reverse the process when I get back. My hope is to bake this "memory" in to the script so that everything just 'feels' the same no matter where I am or what server I'm on. But the linked docs all seem to be focused on setting hot keys to trigger some action. Am I reading the docs correctly or is there an aspect of those that I missed?

**Reply by wez** [reply]
Read https://wezfurlong.org/wezterm/recipes/passing-data.html again.
The approach should be that your shell script sets a user-var prior to launching mosh, then sets it to something else when mosh exits.
You can use the user-var-changed event to notice when that happens and set_config_overrides to update your config accordingly.

**Comment by notbenh** [comment]
Thanks @bew and @wez for helping me understand. I was able to stitch this together from the examples in the docs.

in config:
```
﻿﻿-- == Multiplexing ==
-- Default leader to my standard tmux prefix
config.leader = { key = 'a', mods = 'CTRL' }
-- handler for on the fly switching for to get leader out of the way of remote connections
wezterm.on('user-var-changed', function(window, pane, name, value)
  if name == 'WEZTERM_LEADER_KEY' then
    if string.len(value) > 0 then

      local overrides = window:get_config_overrides() or {}
      overrides.leader = { key = value, mods = 'CTRL' }
      window:set_config_overrides(overrides)

    end
  end
end)

config.keys = {
  { mods = 'LEADER',
    key = 'o',
    action = wezterm.action.ActivatePaneDirection 'Next'
  },
}
```

which allowed me to script the change like I wanted
```
#!/bin/bash
printf "\033]1337;SetUserVar=WEZTERM_LEADER_KEY=%s\007" `echo -n "b" | base64`
mosh dev-server tmux attach
printf "\033]1337;SetUserVar=WEZTERM_LEADER_KEY=%s\007" `echo -n "a" | base64`
```

I'm guessing this is not the most elegant solution but it is working as intended for me. Thanks again @bew and @wez for the pointers.

EDIT: remove typo
